nwb_conversion_tools/datainterfaces/behavior/movie/moviedatainterface.py

```diff
@@ -152,88 +144,126 @@ def run_conversion(
-                            pbar.update(1)
-                    cap.release()
-                    image_series_kwargs.update(
-                        data=H5DataIO(
-                            DataChunkIterator(
-                                tqdm(
-                                    iterable=np.array(mov),
-                                    desc=f"Writing movie data for {Path(file).name}",
-                                    position=tqdm_pos,
-                                    mininterval=tqdm_mininterval,
-                                ),
-                                iter_axis=0,  # nwb standard is time as zero axis
-                                maxshape=tuple(maxshape),
+                    iterable = np.zeros(shape=maxshape, dtype="uint8")
+                    with VideoCaptureContext(str(file)) as video_capture_ob:
+                        if stub_test:
+                            video_capture_ob.frame_count = 10
+                        with tqdm(
+                            desc=f"Reading movie data for {Path(file).name}",
+                            position=tqdm_pos,
+                            total=total_frames,
+                            mininterval=tqdm_mininterval,
+                        ) as pbar:
+                            for n, frame in enumerate(video_capture_ob):
+                                iterable[n, :, :, :] = frame
+                                pbar.update(1)
+                    data = H5DataIO(
+                        DataChunkIterator(
+                            tqdm(
+                                iterable=iterable,
+                                desc=f"Writing movie data for {Path(file).name}",
+                                position=tqdm_pos,
+                                mininterval=tqdm_mininterval,
                             ),
-                            compression="gzip",
-                            chunks=best_gzip_chunk,
-                        )
+                            iter_axis=0,  # nwb standard is time as zero axis
+                            maxshape=maxshape,
+                        ),
+                        compression="gzip",
+                        compression_opts=compression_options,
+                        chunks=best_gzip_chunk,
                     )
+
+                image_series_kwargs.update(data=data)
+                if check_regular_timestamps(ts=timestamps):
+                    image_series_kwargs.update(starting_time=starting_times[j], rate=fps)
+                else:
+                    image_series_kwargs.update(timestamps=timestamps)
+                    if starting_times[j] is None:
+                        current_description = image_series_kwargs.get("description", "")
+                        image_series_kwargs.update(
+                            description=current_description + "Start time not provided. Timestamps may not be accurate."
+                        )
```

Comment:
```suggestion
                    if np.isnan(starting_times[j]):
                        warn("Start time not provided. Timestamps may not be accurate.")
```

TBH people rarely read the descriptions; I would just issue a warning to the person running the conversion to (in addition with the warning above) inform them that timing information might not be accurate.

Reply:
a warning is important for the user doing the conversion. 
But description is important to communicate this to a user in the far future who might access the file. Agreed, Its more likely to be overlooked but its good to still have it in there.

Reply:
In general, I don't think we should automatically adjust object descriptions if specified by the user. We can adjust it as you propose in the case where the default description is being generated, but it's weird that a user would have to specify a data-related argument to prevent text-based metadata from being appended to a manually set one.

Example: Let's say the video spans the entire session duration, and starts exactly at relative time `0.0`. They took the time to write a nice detailed description for the ImageSeries in the metadata, but it didn't occur to the user to have to hard-specify `starting_times=[0.0]` since it seemed obvious to them in this case. They didn't check the output NWBFile super carefully, and missed that we automatically added extra information onto the description. Now everyone that uses the file afterward and does take the time to read the new description in full will question the `timestamps`.

Reply:
then should we just make starting_times as a required argument?

Reply:
> then should we just make starting_times as a required argument? When there are >1 video files to be written?

I thought about that but they could be simultaneous captures of the same experiment (say, from different angles) so that's not fair either. And `starting_time` isn't required on any other interfaces so it would be weird to do that here.

I would just not adjust the description unless we're filling it with a default one.

Reply:
a bit like this? https://github.com/catalystneuro/nwb-conversion-tools/pull/372/commits/0125ee9a5c2bbbd912a9e739e676b2a6a5632f7b

Reply:
IMO, if its a rare case of simultaneous capture, then we should enforce the user to specify the `starting_times`. I think its important info for dataset in the `ImageSeries`.

Reply:
I think its better to be strict about it rather than generate a ton of nwbfiles that dont have this important info no?

Reply:
Sure